Minebot/src/main/java/net/famzangl/minecraft/minebot/ai/path/PlantPathFinder.java

```diff
@@ -208,6 +217,10 @@ protected PathfindingSetting loadSettings(MinebotSettingsRoot settingsRoot) {
 
 	@Override
 	protected float rateDestination(int distance, int x, int y, int z) {
+		System.out.println(world.toString() + " " + distance + " " + x + " " + y + " " + z);
```

Comment:
You use a lot of println to debug.

You can use the logger instead. Use logger.info() when you are developing. Change it to logger.debug() or logger.trace() afterwards.

That way, you can leave this debug output in the game and re-enalbe it easier afterwards.

Reply:
Yes ): println tells me where everything is coming from / lines class etc, and the stack trace. I see though, I am not accustomed to debugger loggers.

Reply:
During development, I like to add logging breakpoints. This allows adding log messages while the game is running and you can easily add a condition to only print e.g. the coordinates that are interesting for you. Make sure to remove the tick next to "Suspend" and to start Minebot in debug mode.
![Bildschirmfoto von 2020-11-07 23-16-07](https://user-images.githubusercontent.com/5981823/98452386-42b19780-214f-11eb-95c4-21bcf579d865.png)